.gitlab-ci.yml

```
@@ -28,27 +28,41 @@ variables:
   S3_BUCKET_NAME: "mender"
   S3_BUCKET_PATH: "mender-artifact"
   DOCKER_BUILDKIT: 1
+  DOCKER_PLATFORMS:
+    value: "linux/amd64,linux/arm64"
+    description: "Platforms for building and distributing container images"
 
 test:check-license-source:
   variables:
     LICENSE_HEADERS_IGNORE_FILES_REGEXP: './artifact/keyfactor/\(signer\|signer_test\)\.go'
 
 build:docker:
-  image: docker
+  image: docker:26-cli
   needs: []
   services:
-    - docker:dind
+    - docker:26-dind
   stage: build
+  before_script:
+    - docker login --username "$CI_REGISTRY_USER" --password "$CI_REGISTRY_PASSWORD" "$CI_REGISTRY"
```

If the password is not masked, this line would leak it. It is better to use

```suggestion
    - echo ${CI_REGISTRY_PASSWORD} | docker login --username ${CI_REGISTRY_USER} --password-stdin ${CI_REGISTRY}
```

---

I'm not sure I get it. Isn't it the same with the echo command? The variable is expanded there as well.

---

It won't be logged into stdout - (iow  the CI logs). I cannot link to an LFF - but search for "GitLab CI secrets leak" in the LFF archive :)